### PR TITLE
Fix missing of CUDA device on non-GPU host

### PR DESCRIPTION
**Pull request by mg0880gm**
The issue that "no CUDA-capable device is detected" occurs when calling MxNet from Java/Scala on a host without GPU. 

The user scenario is to run MxNet model on CPU-only host with Context input as cpu(0). The default GPU unit shares the same index of zero with the default CPU unit. By default the MXNET_USE_CUDA is enabled and the logic will proceed to the line of "CUDA_CALL". The calling of cudaSetDevice exits with exception since there is GPU found for index of zero. The proposal here is to enable the "CUDA_CALL" only when the device type of Context is not CPU.

The whole project with code changes proposed here has been built successfully in local host. After that the running of MxNet model was successful in CPU-only host for evaluation.

### CODE REVIEW

File: scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc [hunk 1]
```diff
@@ -2301,7 +2301,9 @@ JNIEXPORT jint JNICALL Java_ml_dmlc_mxnet_LibInfo_mxCustomOpRegister
                   ptrsArr, (jsize)0, (jsize)size, reinterpret_cast<jlong*>(ptrs));
 #if MXNET_USE_CUDA
                 mxnet::NDArray* tmp = reinterpret_cast<mxnet::NDArray*>(ptrs[0]);
-                CUDA_CALL(cudaSetDevice(tmp->ctx().dev_id));
+                if (tmp->ctx().dev_type != mxnet::Context::kCPU) {
```
Comment (piiswrong) [inline scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc]: == kGPU

**Comment by Ldpe2G** [comment]
@mg0880gm thanks for the pr.

File: scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc (hunk 1 above)
Comment (mg0880gm) [reply scala-package/native/src/main/native/ml_dmlc_mxnet_native_c_api.cc]: Revise the conditional statement to be more readable. Thanks for the comment.

**Comment by mg0880gm** [comment]
After modification of the conditional statement, the CI failed. Actually the local build was successful. Digging into the log of CI test shows it's related to the Python test part. Any suggestion or comment please? @piiswrong @Ldpe2G

**Comment by Ldpe2G** [comment]
try 
```bash
git commit --amend --no-edit && git push --force
```
to relaunch the test

**Comment by mg0880gm** [comment]
Please help review the latest commit @piiswrong @Ldpe2G.

**Comment by Ldpe2G** [comment]
@javelinjs

**Comment by javelinjs** [comment]
Thanks.

**Pull request merged**